src/main/scala/piconot/internal/PicoBot.scala

```diff
@@ -0,0 +1,113 @@
+package piconot.internal
+import picolib.maze.Maze
+import picolib.semantics._
+
+
+  /**
+   * Our rule represents the left half of a picobot rule.  That is it
+   * represents for what initial state this rule applies to and for which
+   * opened or closed NEWS directions the rule applies to.
+   */
+case class OurRule (val state : OurState, val cond : Conditions){
+
+	/**
+	 * helper function to convert strings such as "S" to the move direction South
+	 */
+  def charToMoveDirection (s : String) : MoveDirection=
+	{
+		var goDirection : MoveDirection  = North;
+    if (s.equals("E"))
+      goDirection = East
+    else if (s.equals("W"))
+      goDirection = West
+    else if (s.equals("S"))
+      goDirection = South
+    goDirection
+	}
+  
```

Comment:
As implemented anything that isn't a direction is considered North anyway, so it would be fine.